### PR TITLE
Use new maintained django-cors-headers package

### PR DESCRIPTION
**Pull request by stsewd**
May be blocked by https://github.com/adamchainz/django-cors-headers/issues/830, but we could also just have a single regex for all the allowed URLs.

### CODE REVIEW

**Review by humitos — COMMENTED** [review]
This looks good to me. I'd not block this PR on the setting config. I'd go ahead with the solution you proposed `(url1|url2)`. This string can be generated looping over the current list we already have:

```python
# Generate the regex string manually because a list is not supported
# https://github.com/adamchainz/django-cors-headers/issues/830
CORS_URLS_REGEX = "(" + "|".join(CORS_URLS_ALLOW_ALL_REGEX) + ")"
```

Example,

```
>>> l = ['1', '2', '3', '4']
>>> '(' + '|'.join(l) + ')'
'(1|2|3|4)'
>>>
```

**Comment by stsewd** [comment]
They aren't going to add that option, so back to use a long regex instead. I also hit https://github.com/adamchainz/django-cors-headers/issues/558, but it was more of an issue with our tests, we were trying to test that middleware isolated passing a dictionary as the response, instead of creating a response, I'm just testing the whole request instead.

File: readthedocs/settings/base.py [hunk 1]
```diff
@@ -734,15 +736,17 @@ def DOCKER_LIMITS(self):
...
+    # 
+    # NOTE: We don't use `CORS_ALLOW_ALL_ORIGINS=True`,
+    # since that will set the `Access-Control-Allow-Origin` header to `*`,
+    # we won't be able to pass credentials fo the sustainability API with that value.
+    CORS_ALLOWED_ORIGIN_REGEXES = [re.compile(".+")]
+    CORS_ALLOW_HEADERS = list(default_headers) + [
         'x-hoverxref-version',
-        'x-csrftoken'
-    )
+    ]
```
Comment (humitos) [inline readthedocs/settings/base.py:749]: Just to note we are adding 3 extra allowed headers with this change:

```
    "accept-encoding",
    "dnt",
    "user-agent",
```

**Review by humitos — APPROVED** [review]

File: readthedocs/settings/base.py [hunk 2]
```diff
@@ -751,14 +755,19 @@ def DOCKER_LIMITS(self):
...
     # URLs to allow CORS to read from unauthed.
-    CORS_URLS_ALLOW_ALL_REGEX = [
-        r"^/api/v2/footer_html",
-        r"^/api/v2/search",
-        r"^/api/v2/docsearch",
-        r"^/api/v2/embed",
-        r"^/api/v3/embed",
-        r"^/api/v2/sustainability",
-    ]
+    CORS_URLS_REGEX = re.compile(
```
[the remark this answers is not in the document]
Comment (stsewd) [reply readthedocs/settings/base.py:758]: yeah, that's the type https://docs.python.org/3/library/typing.html#typing.Pattern.

**Pull request merged**